src/util/log/fd_log.c

```diff
@@ -493,7 +493,8 @@ fd_log_private_1( int          level,
     /* 4 */ "ERR    ",
     /* 5 */ "CRIT   ",
     /* 6 */ "ALERT  ",
-    /* 7 */ "EMERG  "
+    /* 7 */ "EMERG  ",
+    /* 8 */ "HEXDUMP"
```

Comment:
No object to notion of a HEXDUMP but wouldn't do it as log level.  Log levels mirror the syslog log levels and trigger various behaviors based on severity (like flushing to stderr / logfile, aborting program and what not).  So a HEXDUMP at level 8 would be above an emergency from a logging POV.

Ideally should be possible to do a HEXDUMP at an INFO or NOTICE or what not level.

Reply:
Thanks very much for the review.

Yeap, I hear you. 

I can just make changes such that hex dumps are done at the INFO level, if this sounds good to you. Or, alternatively, I can the ability to log hex dumps at arbitrary log levels, either a) via macros like `FD_LOG_HEXDUMP_INFO`, `FD_LOG_HEXDUMP_NOTICE`, `FD_LOG_HEXDUMP_WARNING`, etc, or b) by passing in a log level parameter like `FD_LOG_HEXDUMP(FD_LOG_LEVEL_INFO, ...)`, or `FD_LOG_HEXDUMP(FD_LOG_LEVEL_WARNING, ...)`.

Please let me know which you prefer 👍

Reply:
See comments below which address a couple of this with that (along with my tendency to go into ad hoc rants of about issues brokenness of languages and the pain it cause people like us ... it's not directed at your code).

Option a is perfectly cromulent from my POV (make fd_log_hexdump becomes something akin to fd_log_private_1 stuff like mentioned below to deal with the other linguistic unpleasantnesses the logger has to fix up).  Also see the pretty printer in fd_pod_ctl.c mentioned below.  It does a two column hex dump style already:
```  0123: 00 11 22 33 44 55 66 77  88 99 AA BB CC DD EE FF  thequickbrownfox```
So some consolidation / reuse opportunity might be worthwhile.

Reply:
Also related note, if it is ends up making sense to have `fd_log_hexdump` meant for direct use by log users, the behaviors for various permutations of NULL descr, NULL buf and/or 0 sz need to be specified, trapped and/or logged as appropriate.